tests/ContactForm7Test.php

```diff
@@ -0,0 +1,318 @@
+    {
+        global $mocked_actions, $mocked_filters;
+        $mocked_actions = [];
+        $mocked_filters = [];
+        WP_Mock::setUp();
+
+        // Initialize the Forms instance
+        $this->forms = new Forms();
+    }
+
+    protected function tearDown(): void
+    {
+        WP_Mock::tearDown();
+    }
+
+    public function testSetupActions()
+    {
+        // Call the 'setup' method on the 'Forms' instance.
+        $this->forms->setup();
+        // Declares a global variable $mocked_actions, which is used to store the actions
+        global $mocked_actions;
+        // Asserts that the 'wp_enqueue_scripts' action is registered with the correct callback and priority.
+        $this->assertContains([
+            'hook' => 'wp_enqueue_scripts',
+            'callback' => [AdCaptcha::class, 'enqueue_scripts'],
+            'priority' => 9,
+            'accepted_args' => 1
+        ], $mocked_actions);
+
+        $this->assertContains([
+            'hook' => 'wp_enqueue_scripts',
+            'callback' => [$this->forms, 'block_submission'],
+            'priority' => 9,
+            'accepted_args' => 1
+        ], $mocked_actions);
+
+        $this->assertContains([
+            'hook' => 'wp_enqueue_scripts',
+            'callback' => [$this->forms, 'get_success_token'],
+            'priority' => 9,
+            'accepted_args' => 1
+        ], $mocked_actions);
+
+        $this->assertContains([
+            'hook' => 'wp_enqueue_scripts',
+            'callback' => [$this->forms, 'reset_captcha_script'],
+            'priority' => 9,
+            'accepted_args' => 1
+        ], $mocked_actions);
+    }
+
+    public function testSetupFilters()
+    {
+        $this->forms->setup();
+
+        global $mocked_filters;
+
+        $this->assertContains([
+            'hook' => 'wpcf7_form_elements',
+            'callback' => [$this->forms, 'captcha_trigger_filter'],
+            'priority' => 20,
+            'accepted_args' => 1
+        ], $mocked_filters);
+
+        $this->assertContains([
+            'hook' => 'wpcf7_form_hidden_fields',
+            'callback' => [$this->forms, 'add_adcaptcha_response_field'],
+            'priority' => 10,
+            'accepted_args' => 1
+        ], $mocked_filters);
+
+        $this->assertContains([
+            'hook' => 'wpcf7_spam',
+            'callback' => [$this->forms, 'verify'],
+            'priority' => 9,
+            'accepted_args' => 1
+        ], $mocked_filters);
+    }
+
+    public function testVerifySpamAlreadyTrue()
+    {
+        $result = $this->forms->verify(true);
+        $this->assertTrue($result);
+    }
+
+    public function testCaptchaTriggerFilter()
+    {
+        // Sample HTML input
+        $inputHtml = '<form>
+                        <input type="text" name="name">
+                        <button type="submit">Submit</button>
+                      </form>';
+
+        // Mock the expected output from ob_captcha_trigger
```

Comment:
What's `ob_captcha_trigger`?

Reply:
THis function is located in the Widget folder AdCatpcha.php. OB stand for output buffering. All output sent from PHP (such as HTML or text) is captured and stored in an internal buffer instead of being sent to the browser immediately